liquibase-core/src/main/java/liquibase/snapshot/jvm/IndexSnapshotGenerator.java

```diff
@@ -266,7 +266,7 @@ protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot
                     foundIndexes.put(correctedIndexName, returnIndex);
                 }
 
-                if ((database instanceof MSSQLDatabase) && (Boolean) row.get("IS_INCLUDED_COLUMN")) {
+                if ((database instanceof MSSQLDatabase) && Boolean.valueOf(row.get("IS_INCLUDED_COLUMN").booleanValue()) {
```

Comment:
`Boolean#valueOf` returns a `Boolean` at any time as described in https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Boolean.html#valueOf(java.lang.String)

> Returns a Boolean with a value represented by the specified string. The Boolean returned represents a true value if the string argument is not null and is equal, ignoring case, to the string "true". Otherwise, a false value is returned, including for a null argument.

If the return value is `null`, `Boolean#valueOf` returns `Boolean.FALSE`.

Reply:
Thanks, but I think the valueOf(string) isn't being called because of your `.booleanValue()` call on there?

But if you agree the ObjectUtil version fixes the problem, it's probably the better one regardless. If you could update the PR that would be great, otherwise we can push a change as part of the next review round.

Reply:
It is called. `.booleanValue()` will be called on the result of `valueOf(...)`. If u remove the `.booleanValue()` the JVM will call it for unboxing.
But anyway it is your code and if u feel more comfortable with the `ObjectUtil` solution, I'll use this.